source/core/src/main/com/csse3200/game/screens/AnimalSelectionScreen.java

```diff
@@ -0,0 +1,184 @@
+package com.csse3200.game.screens;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.ScreenAdapter;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.*;
+import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;
```

Comment:
This file is a bit too long and does too many things compared to what a screen should do imo. See how main menu is implemented - there is a component for main menu display and actions and the main menu screen simply links those by registering with the service locator. The animal selection screen should be implemented in a similar manner I think.